server/src/handlers/http/rbac.rs

```diff
@@ -0,0 +1,140 @@
+use crate::{
+    option::CONFIG,
+    rbac::{
+        get_user_map,
+        user::{PassCode, User},
+    },
+    storage::{self, ObjectStorageError, StorageMetadata},
+    validator::{self, error::UsernameValidationError},
+};
+use actix_web::{http::header::ContentType, web, Responder};
+use http::StatusCode;
+use tokio::sync::Mutex;
+
+// async aware lock for updating storage metadata and user map atomicically
+static UPDATE_LOCK: Mutex<()> = Mutex::const_new(());
+
+pub async fn put_user(username: web::Path<String>) -> Result<impl Responder, RBACError> {
```

Comment:
Add a detailed comment here on how the API behaves - Takes username, creates the user, responds with a password. That password can't be accessed again, to change password call reset and so on.